setup.py

```diff
@@ -28,8 +28,10 @@ def _cryptography_version():
     # https://cryptography.io/en/latest/changelog/#v2-5
     if platform.python_implementation() == 'PyPy' and platform.python_version() < '5.4':
         return 'cryptography < 2.5'
+    elif platform.python_implementation() == 'CPython' and platform.python_version() < '3.6':
+        return 'cryptography < 3.4'
```

Comment:
I don't think this is necessary.

Pip will resolve the correct version already based on `cryptography`'s `python_requires`

```
$ python --version
Python 3.5.9
$ pip --version
pip 9.0.1
$ pip install python-jose[cryptography]
...
Installing collected packages: pyasn1, rsa, six, ecdsa, pycparser, cffi, cryptography, python-jose
Successfully installed cffi-1.14.5 cryptography-3.2.1 ecdsa-0.14.1 pyasn1-0.4.8 pycparser-2.20 python-jose-3.2.0 rsa-4.7 six-1.15.0
```

Note that on Python 3.5, even with an old version of pip, resolves `cryptography` to be `3.2.1` instead of `3.3.x` or `3.4.x`

Reply:
Ah I see, good to know. I will drop this part then.